tools/initial_ros_setup.bash

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+
+ENABLE_ROS1=true
+ENABLE_BUILD=true
+for opt in "$@" ; do
+	case $opt in
+		--no-ros1)
+			ENABLE_ROS1=false
+			shift
+		;;
+		--download-only)
+			ENABLE_BUILD=false
+			shift
+		;;
+		*)
+			echo "Invalid option: $opt"
+			echo "Valid options:"
+			echo "--no-ros1       Disables downloading and building ROS1 dependencies and ROS1 bridge"
+			echo "--download-only Skips the build step and only downloads the code"
+			exit 1
+		;;
+	esac
+done
+
+set -e
+CHECKPOINT_FILES=''
+
+CWD=`pwd`
+function return_to_root_dir {
+	cd $CWD
+}
+
+function download_navstack {
+	echo "Downloading the ROS 2 navstack"
+	if [ -f "custom_nav2.repos" ]; then #override default location for testing
+		vcs-import < custom_nav2.repos
+	else
+		git clone https://github.com/ros-planning/navigation2.git
+	fi
+	return_to_root_dir
+}
+
+function download_ros2 {
+	echo "Downloading ROS 2 Release Latest"
+	mkdir -p ros2_ws/src
+	cd ros2_ws
+	wget https://raw.githubusercontent.com/ros2/ros2/release-latest/ros2.repos
+	vcs import src < ros2.repos
+	return_to_root_dir
+}
+
+function download_ros2_dependencies {
+	echo "Downloading the dependencies workspace"
+	mkdir -p navstack_dependencies_ws/src
+	cd navstack_dependencies_ws
+	vcs import src < ${CWD}/navigation2/tools/ros2_dependencies.repos
+	return_to_root_dir
+}
+
+function download_ros1_dependencies {
+	echo "Downloading the ROS 1 dependencies workspace"
+	mkdir -p ros1_dependencies_ws/src
+	cd ros1_dependencies_ws
+	vcs import src < ${CWD}/navigation2/tools/ros1_dependencies.repos
+	return_to_root_dir
+}
+
+function checkpoint {
+	local CHECKPOINT_FILE_NAME=.INITIAL_SETUP_$1
+	CHECKPOINT_FILES="${CHECKPOINT_FILES} ${CHECKPOINT_FILE_NAME}"
+	if [ ! -f ${CHECKPOINT_FILE_NAME} ]; then
+		$1
+		touch ${CHECKPOINT_FILE_NAME}
+	fi
+}
+
+function download_all {
+	checkpoint download_navstack
+	checkpoint download_ros2_dependencies
+	checkpoint download_ros2
+	if [ "$ENABLE_ROS1" == true ]; then
+		checkpoint download_ros1_dependencies
+	fi
+}
+
+echo "This script will download the ROS 2 latest release workspace, the"
+echo "dependencies workspace and the ros_navstack_port workspace to the"
+echo "current directory and then build them all. There should be no ROS"
+echo "environment variables set at this time."
+echo
+echo "The current directory is $CWD"
+echo
+read -p "Are you sure you want to continue? [yN]" -n 1 -r
```

Comment:
For this to work in automation, need a flag to skip the "Are you sure..." check

Reply:
Going with the "yes | " option.